csrc/executor_kernel_arg.h

```diff
@@ -62,14 +62,10 @@ struct TensorArgCodegen {
   using data_type = T;
   using index_type = nvfuser_index_t;
   static constexpr int ndims = N;
-
-  T& operator[](nvfuser_index_t ind) {
-    return data[ind];
-  };
-
-  T* data;
+  size_t data_ptr_address = 0;
```

Comment:
we are changing the structure here, but I don't see changes applied to the `./runtime/tensor.cu`.
maybe I missed something obvious here, but I don't see how this is working...

Reply:
I guess my question here is, why are we using `size_t` instead of `T*` for `data_ptr_address`?

Reply:
When the `at::Tensor` had `meta` device, the `data_ptr` is `nullptr`. It seemed unwise to have access to the raw pointer if we're only using the `size_t` value in the codegen.

The pointer size of `T*` is the same as `size_t`, so `TensorArgCodegen` should be the same as `Tensor`.

Reply:
This makes me very nervous here. :nerd_face: I think what you are saying is generally true for us now. But not guaranteed across platforms.

> When the at::Tensor had meta device, the data_ptr is nullptr. It seemed unwise to have access to the raw pointer if we're only using the size_t value in the codegen.

That's just a nullptr. I don't see how it's different from a `0` in a size_t. I'd argue a nullptr is less ambiguous.
What I'm really saying is that I don't think there's any real benefit to store a `size_t` instead of a pointer here. I know out-side of the kernel we are only accessing it as a value for alignment checks, but we can always just cast it in those cases and not risking on the assumption of `sizeof(T*) == sizeof(size_t)`.